pkg/operator/controller/externaldns/status.go

```diff
@@ -0,0 +1,285 @@
+	for _, cond := range deployment.Status.Conditions {
+		if cond.Type == appsv1.DeploymentAvailable {
+			switch cond.Status {
+			case corev1.ConditionFalse:
+				return metav1.Condition{
+					Type:               ExternalDNSDeploymentAvailableConditionType,
+					Status:             metav1.ConditionFalse,
+					Reason:             "DeploymentUnavailable",
+					Message:            fmt.Sprintf("The deployment has Available status condition set to False (reason: %s) with message: %s", cond.Reason, cond.Message),
+					LastTransitionTime: cond.LastUpdateTime,
+				}
+			case corev1.ConditionTrue:
+				return metav1.Condition{
+					Type:               ExternalDNSDeploymentAvailableConditionType,
+					Status:             metav1.ConditionTrue,
+					Reason:             "DeploymentAvailable",
+					Message:            "The deployment has Available status condition set to True",
+					LastTransitionTime: cond.LastUpdateTime,
+				}
+			}
+			break
+		}
+	}
+	return metav1.Condition{
+		Type:               ExternalDNSDeploymentAvailableConditionType,
+		Status:             metav1.ConditionUnknown,
+		Reason:             "DeploymentAvailabilityUnknown",
+		Message:            "The deployment has no Available status condition set",
+		LastTransitionTime: metav1.NewTime(clock.Now()),
+	}
+
+}
+func computeMinReplicasCondition(deployment *appsv1.Deployment) metav1.Condition {
+	replicas := int32(1)
+	if deployment.Spec.Replicas != nil {
+		replicas = *deployment.Spec.Replicas
+	}
+
+	pointerTo := func(val intstr.IntOrString) *intstr.IntOrString { return &val }
+	maxUnavailableIntStr := pointerTo(intstr.FromString("25%"))
+	maxSurgeIntStr := pointerTo(intstr.FromString("25%"))
+	if deployment.Spec.Strategy.Type == appsv1.RollingUpdateDeploymentStrategyType && deployment.Spec.Strategy.RollingUpdate != nil {
+		if deployment.Spec.Strategy.RollingUpdate.MaxUnavailable != nil {
+			maxUnavailableIntStr = deployment.Spec.Strategy.RollingUpdate.MaxUnavailable
+		}
+		if deployment.Spec.Strategy.RollingUpdate.MaxSurge != nil {
+			maxSurgeIntStr = deployment.Spec.Strategy.RollingUpdate.MaxSurge
+		}
+	}
+	maxSurge, err := intstr.GetScaledValueFromIntOrPercent(maxSurgeIntStr, int(replicas), true)
+	if err != nil {
+		return metav1.Condition{
+			Type:    ExternalDNSDeploymentReplicasMinAvailableConditionType,
+			Status:  metav1.ConditionUnknown,
+			Reason:  "InvalidMaxSurgeValue",
+			Message: fmt.Sprintf("invalid value for max surge: %v", err),
+		}
+	}
+	maxUnavailable, err := intstr.GetScaledValueFromIntOrPercent(maxUnavailableIntStr, int(replicas), false)
+	if err != nil {
+		return metav1.Condition{
+			Type:    ExternalDNSDeploymentReplicasMinAvailableConditionType,
+			Status:  metav1.ConditionUnknown,
+			Reason:  "InvalidMaxUnavailableValue",
+			Message: fmt.Sprintf("invalid value for max unavailable: %v", err),
+		}
+	}
+	if maxSurge == 0 && maxUnavailable == 0 {
+		//TODO understand why this is the default
```

Comment:
This is actually a blocking situation when you have both `maxSurge` and `maxUnavailable` set to 0. `maxSurge=0` says "do the in-place upgrade" (don't spawn new PODs, replace existing ones) while `maxUnavailable=0` says "and please don't make any POD unavailable". I have to try but I think that the rolling update with these values will never pass and will timeout.

So, some minimal number was still added.